org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/preferences/internal/DefaultPersistentModel.java

```diff
@@ -36,6 +41,9 @@
 
     private static final String PROPERTY_SUBPROJECTS = "subprojectPaths";
     private static final String PROPERTY_BUILD_DIR = "buildDir";
+    private static final String PROPERTY_CLASSPATH = "classpath";
+    private static final String PROPERTY_DERIVED_RESOURCES = "derivedResources";
+    private static final String PROPERTY_LINKED_RESOURCES = "linkedResources";
 
     private final IProject project;
     private final Map<String, String> entries;
```

Comment:
I think we should only serialize/deserialize on startup/teardown and keep the results in memory otherwise. I.e. don't have a single Map field, but actually have fields for classpath, derived resources etc.